### ISSUE TITLE
udev rules with $env{INTERFACE} broken, if interface is renamed

### ISSUE THREAD

**Issue by haraldh** [systemd/systemd#14935]
https://github.com/systemd/systemd/commit/1448820aadd138882232f169157de133a8f407b0

broke this rule:
```
ACTION=="add", SUBSYSTEM=="net", RUN+="/sbin/initqueue --name ifup-$env{INTERFACE} --unique --onetime /sbin/ifup $env{INTERFACE}"
```

If an interface is renamed, `$env{INTERFACE}` is changed in `src/libsystemd/sd-device/device-private.c:device_rename()`

https://github.com/systemd/systemd/blob/1ad448673ed3bfb0f989909485f01f485888d90d/src/libsystemd/sd-device/device-private.c#L629-L639

With the offending patch, `$env{INTERFACE}` is evaluated with the old interface name

**Comment by haraldh** [comment]
This broke dracut in legacy network mode.

**Comment by yuwata** [comment]
Hmm. https://github.com/systemd/systemd/pull/12416#issuecomment-487616695

**Comment by yuwata** [comment]
What happens when $name is used instead of $env{INTERFACE}?

**Comment by mrc0mmand** [comment]
> What happens when `$name` is used instead of `$env{INTERFACE}`?

That seems to work. Tried `TEST-20-NFS` with `network-legacy` and this patch:

```diff
diff --git a/modules.d/35network-legacy/net-genrules.sh b/modules.d/35network-legacy/net-genrules.sh
index d531bb56..41ee93fb 100755
--- a/modules.d/35network-legacy/net-genrules.sh
+++ b/modules.d/35network-legacy/net-genrules.sh
@@ -65,9 +65,9 @@ command -v fix_bootif >/dev/null || . /lib/net-lib.sh
         bootdev=$(cat /tmp/net.bootdev)
     fi
 
-    ifup='/sbin/ifup $env{INTERFACE}'
+    ifup='/sbin/ifup $name'
 
-    runcmd="RUN+=\"/sbin/initqueue --name ifup-\$env{INTERFACE} --unique --onetime $ifup\""
+    runcmd="RUN+=\"/sbin/initqueue --name ifup-\$name --unique --onetime $ifup\""
 
     # We have some specific interfaces to handle
     if [ -n "${RAW_IFACES}${IFACES}" ]; then
```

and the network came up as expected in all cases.

**Comment by yuwata** [comment]
@mrc0mmand Thank you for the testing.

So, I'd like to keep the current behavior of udevd.
- The commit fixes a bug explained in #12291; `$result` is overwritten by the last `PROGRAM=`.
- The current behavior is intuitive and easy to understand; each line is evaluated sequentially.
- The previous behavior is not documented.

**Comment by mrc0mmand** [comment]
@yuwata thanks for the explanation. Fix was merged into dracut's master (https://github.com/dracutdevs/dracut/commit/a8ba1c4e25051c3c482bb7a6a754c9d785463917), so I guess we can close this issue.
